GAE/src/com/gallatinsystems/gis/map/MapUtils.java

```diff
@@ -174,6 +175,11 @@ public static void recomputeCluster(Cache cache, SurveyedLocale locale, int delt
             }
         }
 
+        // delete locale if the Delta was a subtraction
+        SurveyedLocaleDao slDao = new SurveyedLocaleDao();
+        if (delta < 0) {
```

Comment:
This code is aware of that? The comment states:

     // delta can be +1 or -1, depending on if we want to add a point or delete a point.

So `delta` can be `-1`, but a `SurveyedLocale` can have more `SurveyInstance`s ?